router/core/src/main/scala/io/buoyant/router/RoutingFactory.scala

```diff
@@ -97,19 +101,24 @@ class RoutingFactory[Req, Rsp](
       }
 
       for {
-        (dst, req1) <- getDst(req0).rescue {
-          case e: Throwable =>
-            Annotations.Failure.Identification.record(e)
-            Future.exception(UnknownDst(req0, e))
+        identified <- getDst(req0).transform {
```

Comment:
tioli: if you have unapply, you can sugar this with `IdentifiedRequest(dst, req) <- getDst(req0).transform { ...`